core/src/main/java/org/apache/gravitino/storage/relational/service/ModelVersionMetaService.java

```diff
@@ -0,0 +1,278 @@
+    }
+
+    List<ModelVersionAliasRelPO> aliasRelPOs =
+        SessionUtils.getWithoutCommit(
+            ModelVersionAliasRelMapper.class,
+            mapper -> {
+              if (isVersionNumber) {
+                return mapper.selectModelVersionAliasRelsByModelIdAndVersion(
+                    modelEntity.id(), Integer.valueOf(ident.name()));
+              } else {
+                return mapper.selectModelVersionAliasRelsByModelIdAndAlias(
+                    modelEntity.id(), ident.name());
+              }
+            });
+
+    return POConverters.fromModelVersionPO(modelIdent, modelVersionPO, aliasRelPOs);
+  }
+
+  public void insertModelVersion(ModelVersionEntity modelVersionEntity, boolean overwrite)
+      throws IOException {
+    if (overwrite) {
+      LOG.warn(
+          "Overwrite is not supported for model version meta, ignoring this flag and "
+              + "inserting the new model version.");
+    }
+
+    NameIdentifier modelIdent = modelVersionEntity.modelIdentifier();
+    NameIdentifierUtil.checkModel(modelIdent);
+
+    // Will throw a NoSuchEntityException if the model does not exist.
+    ModelPO modelPO = ModelMetaService.getInstance().getModelPOByIdentifier(modelIdent);
+    Integer newVersion = modelPO.getModelLatestVersion();
+    ModelVersionEntity entityWithNewVersion =
+        ModelVersionEntity.builder()
+            .withVersion(newVersion)
+            .withModelIdentifier(modelIdent)
+            .withAliases(modelVersionEntity.aliases())
+            .withComment(modelVersionEntity.comment())
+            .withUri(modelVersionEntity.uri())
+            .withProperties(modelVersionEntity.properties())
+            .withAuditInfo(modelVersionEntity.auditInfo())
+            .build();
+
+    ModelVersionPO.Builder builder = ModelVersionPO.builder();
+    builder
+        .withModelId(modelPO.getModelId())
+        .withMetalakeId(modelPO.getMetalakeId())
+        .withCatalogId(modelPO.getCatalogId())
+        .withSchemaId(modelPO.getSchemaId());
+    ModelVersionPO modelVersionPO =
+        POConverters.initializeModelVersionPO(entityWithNewVersion, builder);
+
+    List<ModelVersionAliasRelPO> aliasRelPOs =
+        POConverters.initializeModelVersionAliasRelPO(entityWithNewVersion, modelPO.getModelId());
+
+    try {
+      SessionUtils.doMultipleWithCommit(
+          () ->
+              SessionUtils.doWithoutCommit(
+                  ModelVersionMetaMapper.class,
+                  mapper -> mapper.insertModelVersionMeta(modelVersionPO)),
+          () -> {
+            if (aliasRelPOs.isEmpty()) {
+              return;
+            }
+
+            SessionUtils.doWithoutCommit(
+                ModelVersionAliasRelMapper.class,
+                mapper -> mapper.insertModelVersionAliasRels(aliasRelPOs));
+          },
+          () ->
+              // If the model version is inserted successfully, update the model latest version.
+              SessionUtils.doWithoutCommit(
```

Comment:
According to the current logic, this update operation will succeed under normal circumstances. Should we add a specified version when updating as mentioned above and then verify the update result?

Reply:
Let me see how to handle this.